content/guest-post/nestjs-authentication-with-loginradius-api/index.md

```diff
@@ -0,0 +1,452 @@
+export class AuthService {
+    async signup(registerUserDto: UserDto) {
+      console.log('sign up')
+    }
+}
+```
+
+Note that you are also importing the user DTO and the loginradius-sdk at the top of the file.
+To execute the signup method in the signup route, inject it in the `auth.contoller.ts` file.
+
+```JavaScript
+import { Injectable } from '@nestjs/common';
+import { UserDto } from './dto/user.dto';
+import { AuthService } from './auth.service';
+
+@Injectable()
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) { }
+
+  @Post('signup')
+  async signup(@Body() registerUserDto: UserDto) {
+    let response = await this.authService.signup(registerUserDto)
+    return response
+  }
+}
+```
+
+Before registering the user, validate if the email is already in use.
+
+Start by configuring loginradius-sdk.
+
+```JavaScript
+import * as LRAuthPrrovider from 'loginradius-sdk'
+
+let config = {
+    apiDomain: "api.loginradius.com",
+    apiKey: process.env.API_KEY,
+    apiSecret: process.env.API_SECRET,
+    siteName: process.env.APP_NAME,
+    apiRequestSigning: false,
+    proxy: {
+      host: "",
+      port: "",
+      user: "",
+      password: "",
+    },
+  };
+let lrv2 = LRAuthPrrovider(config);
+
+let sott = process.env.SOTT
+```
+
+Next, check if the email is already in use.
+
+```JavaScript
+@Injectable()
+export class AuthService {
+  async signup(registerUserDto: UserDto) {
+    try {
+      const response = await lrv2.authenticationApi
+        .checkEmailAvailability(registerUserDto.email)
+        if (response.isExist) {
+          return "Email already in use"
+        }
+    catch(error) {
+      return error
+    }
+  }
+}
+```
+
+If the email is not already in use, register the user.
+
+```JavaScript
+@Injectable()
+export class AuthService {
+    async signup(registerUserDto: UserDto) {
+      try {
+        // check if email is already in use
+        const response = await lrv2.authenticationApi
+            .checkEmailAvailability(registerUserDto.email)
+          if (response.isExist) {
+          return "Email already in use"
+          }
+          // create registration model
+          let authUserRegistrationModel = {
+            email: [
+              {
+                type: "primary",
+                value: registerUserDto.email,
+              },
+            ],
+            password: registerUserDto.password,
+          };
+          // register user
+          let user = await lrv2.authenticationApi
+          .userRegistrationByEmail(authUserRegistrationModel, sott)
+          if(user) {
+            return "Sign up successful"
+          }
+        } catch (error) {
+          return error
+        }
+    }
+}
+```
+
+In the above code, you register a new user by passing in the user data to the authentication API. The `authUserRegistrationModel` object defines how the email and password will be stored in the database.
+
+## Log in the User
+
```

Comment:
In the Login Section, you can mention the code that needs to be injected in the `auth.controller.ts` file for the login method to work on login route, similar to as mentioned in the signup section.